### PR TITLE
fixing deps in old lutin packages 

### PR DESCRIPTION
**Pull request by jahierwan**
follow-up to https://github.com/ocaml/opam-repository/pull/19031

### CODE REVIEW

File: packages/lutin/lutin.2.56/opam [hunk 1]
```diff
@@ -53,6 +41,6 @@ Lutin is a language to model stochastic reactive systems. It has been
 initially designed to model environments and perform automated testing
 of reactive systems with Lurette."""
 url {
-  src: "http://www-verimag.imag.fr/DIST-TOOLS/SYNCHRONE/pool/lutin.2.56.tgz"
-  checksum: "md5=73384160b3e71492a663962d262ba8e3"
+  src: "http://www-verimag.imag.fr/DIST-TOOLS/SYNCHRONE/pool/lutin.2.56-fix.tgz"
```
Comment (mseri) [inline packages/lutin/lutin.2.56/opam:45]: This change is fine, the two packages differ just by:
```diff
+ BuildDepends:·str,unix,num,dynlink,extlib,lutils·(>=·1.9),lutin-utils,ezdl,gbddml,polka,camlidl,gmp,rdbg-plugin·(>=·1.177),lustre-v6,lutin,stdlib-shims
- BuildDepends:·str,unix,num,dynlink,extlib,lutils·(>=·1.9),lutin-utils,ezdl,gbddml,polka,camlidl,gmp,rdbg-plugin·(>=·1.177),lustre-v6,lutin
```

File: packages/lutin/lutin.2.70.4/opam [hunk 2]
```diff
@@ -28,29 +30,15 @@ depends: [
...
-  ["perl" "gmp-devel" "mpfr-devel" "m4"] {os-distribution = "centos"}
-  ["perl" "libgmp-dev" "libmpfr-dev" "m4"] {os-family = "debian"}
-]
 dev-repo:
   "git+https://gricad-gitlab.univ-grenoble-alpes.fr/verimag/synchrone/lutin.git"
 url {
-  archive: "http://www-verimag.imag.fr/DIST-TOOLS/SYNCHRONE/pool/lutin.2.70.4.tgz"
-  checksum: "md5=5e27b8a93f33e622e384263d2e85505b"
+  archive: "http://www-verimag.imag.fr/DIST-TOOLS/SYNCHRONE/pool/lutin.2.70.4-fix.tgz"
+  checksum: "md5=3b8375658938b3757ab2a14e5a54a51a"
```
Comment (mseri) [inline packages/lutin/lutin.2.70.4/opam:44]: As above, with an extra fix on the version in the oasis file. Good as well

File: packages/lutin/lutin.2.56/opam (hunk 1 above)
Comment (mseri) [reply packages/lutin/lutin.2.56/opam:45]: And the related `package(stdlib-shims)` directives

**Comment by jahierwan** [comment]
Should I add a dependency to avoid ocaml 4.03?

**Comment by mseri** [comment]
I don't know why it is failing there to be honest

**Comment by jahierwan** [comment]
Ditto...

File: packages/lutin/lutin.2.56/opam [hunk 3]
```diff
@@ -16,7 +16,7 @@ remove: [
   ["ocaml" "%{etc}%/Lutin/setup.ml" "-C" "%{etc}%/Lutin" "-uninstall"]
 ]
 depends: [
-  "ocaml" {>= "4.02" & < "4.12"}
+  "ocaml" {>= "4.02" & < "4.12" & != "4.03"}
```
Comment (mseri) [inline packages/lutin/lutin.2.56/opam:19]: I disagree with this change. The failure is due to some incorrect lower bound in one of the dependencies. The package works fine on 4.03 in general

Comment (mseri) [reply packages/lutin/lutin.2.56/opam:19]:
```suggestion
  "ocaml" {>= "4.02" & < "4.12"}
```

Comment (jahierwan) [reply packages/lutin/lutin.2.56/opam:19]: ok

**Comment by kit-ty-kate** [comment]
Ah, this one is the classic issue with ocamlbuild 0.9.0. I'll add the appropriate constraint.

**Comment by jahierwan** [comment]
Thank you kate!

**Comment by mseri** [comment]
Super, thanks a lot

**Pull request merged**